lib/store.ml

```diff
@@ -21,13 +21,15 @@ end
 module type S =
 sig
   type t
+  type 'a merge_field = 'a option -> 'a option -> 'a option
   type 'a field
   val empty : t
   val set : t -> 'a field -> 'a -> t
   val get : t -> 'a field -> 'a option
   val remove : t -> 'a field -> t
   val merge : t -> t -> t
-  val field : unit -> 'a field
+  val field : 'a merge_field -> 'a field
```

Comment:
I'd probably take inspiration from the `Dyn` module for the string implementation. Even maybe reusing it.